pkg/cvo/availableupdates.go

```diff
@@ -159,13 +171,37 @@ func calculateAvailableUpdatesStatus(clusterID, upstream, channel, version strin
 	}
 }
 
-func checkForUpdate(clusterID, upstream, channel string, currentVersion semver.Version) ([]cincinnati.Update, error) {
+func checkForUpdate(clusterID, upstream, channel string, currentVersion semver.Version, proxyURL *url.URL) ([]cincinnati.Update, error) {
 	uuid, err := uuid.Parse(string(clusterID))
 	if err != nil {
 		return nil, err
 	}
+
 	if len(upstream) == 0 {
 		return nil, fmt.Errorf("no upstream URL set for cluster version")
 	}
-	return cincinnati.NewClient(uuid).GetUpdates(upstream, channel, currentVersion)
+	return cincinnati.NewClient(uuid, proxyURL).GetUpdates(upstream, channel, currentVersion)
+}
+
+// getHTTPSProxyURL returns a url.URL object for the configured
+// https proxy only. It can be nil if does not exist or there is an error.
+func (optr *Operator) getHTTPSProxyURL() (*url.URL, error) {
+	proxy, err := optr.proxyLister.Get("cluster")
+
+	if errors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
```

Comment:
why do you need the `else if` ?

Reply:
I was thinking we would want to know what the error was if it was anything except `IsNotFound`

Reply:
errros.IsNotFound only returns true when the err is not-found..
ie if err is nil, it will not be true.. and if err is not-nil and but not not-found, it will not be true..

so
```
if errors.IsNotFound(err) {
  return nil, nil
}
if err != nil {
  return nil, err
}
```

cover's all the cases...